modules/test-runner/src/swap/swap.test.ts

```diff
@@ -130,4 +130,123 @@ describe("Swaps", () => {
     );
     expect(postSwapFreeBalanceTokenNode).toBeBigNumberEq(swapAmount);
   });
+
+  test("Bot A tries to swap with invalid from token address", async () => {
+    // client deposit and request node collateral
+    await clientA.deposit({ amount: parseEther("0.01").toString(), assetId: AddressZero });
+    await clientA.requestCollateral(tokenAddress);
+
+    const swapRate = await clientA.getLatestSwapRate(AddressZero, tokenAddress);
+    const wrongAddress = "0xdeadbeef";
+    const swapAmount = One;
+    const swapParams: SwapParameters = {
+      amount: swapAmount.toString(),
+      fromAssetId: wrongAddress,
+      swapRate,
+      toAssetId: tokenAddress,
+    };
+    await expect(clientA.swap(swapParams)).rejects.toThrowError(`is not a valid eth address`);
+  });
+
+  test("Bot A tries to swap with invalid to token address", async () => {
+    // client deposit and request node collateral
+    await clientA.deposit({ amount: parseEther("0.01").toString(), assetId: AddressZero });
+    await clientA.requestCollateral(tokenAddress);
+
+    const swapRate = await clientA.getLatestSwapRate(AddressZero, tokenAddress);
+    const wrongAddress = "0xdeadbeef";
+    const swapAmount = One;
```

Comment:
we should make the balances more realistic

Reply:
Why? It shouldn't make a difference, and the one wei helps us catch potential rounding errors

Reply:
I can change it for the some of the other tests to add some variety

Reply:
It really shouldn't make any difference here

Reply:
I'd even argue that we should be testing smaller amounts to catch potential rounding issues